test/acceptance/coffee/CRUDTests.coffee

```diff
@@ -0,0 +1,201 @@
+				beforeEach (done) =>
+					TagsClient.createTag @user_id_1, @tag_1, (@err2, @res2, @body2) =>
+						done()
+
+				it 'should not see an error too', =>
+					expect(@err2).to.not.exist
+
+				it 'should return 200 too', =>
+					expect(@res2.statusCode).to.equal(200)
+
+				it 'should return the same doc', =>
+					expect(@body).to.deep.equal(@body2)
+
+			describe 'creating a second tag', =>
+				beforeEach (done) =>
+					TagsClient.createTag @user_id_1, @tag_2, (@err2, @res2, @body2) =>
+						done()
+
+				it 'should not see an error too', =>
+					expect(@err2).to.not.exist
+
+				it 'should return 200 too', =>
+					expect(@res2.statusCode).to.equal(200)
+
+				it 'should return different doc', =>
+					expect(@body).to.not.deep.equal(@body2)
+
+	describe 'Get', =>
+		beforeEach (done) =>
+			TagsClient.createTag @user_id_1, @tag_1, =>
+				TagsClient.createTag @user_id_1, @tag_2, done
+
+		describe 'getUserTags', =>
+			it 'should return the two tags', (done) =>
+				TagsClient.getUserTags @user_id_1, (err, res, body) =>
+					tags = body.map((tagDoc) -> tagDoc.name).sort()
+					expect(tags).to.deep.equal([@tag_1, @tag_2])
+					done()
+
+			describe 'with a second user', =>
+				beforeEach (done) =>
+					TagsClient.createTag @user_id_2, @tag_3, done
+
+				it 'should show two tags for user 1', (done) =>
+					TagsClient.getUserTags @user_id_1, (err, res, body) =>
+						tags = body.map((tagDoc) -> tagDoc.name).sort()
+						expect(tags).to.deep.equal([@tag_1, @tag_2])
+						done()
+
+				it 'should show one tag to user 2', (done) =>
+					TagsClient.getUserTags @user_id_2, (err, res, body) =>
+						tags = body.map((tagDoc) -> tagDoc.name)
+						expect(tags).to.deep.equal([@tag_3])
+						done()
+
+	describe 'Update', =>
+		beforeEach (done) =>
+			TagsClient.createTag @user_id_1, @tag_1, (err, res, body) =>
+				@user_1_tag_1_id = body._id
+				TagsClient.createTag @user_id_1, @tag_2, done
+
+		describe 'updateTagUserIds', =>
+			beforeEach (done) =>
+				TagsClient.updateTagUserIds @user_id_1, @user_id_2, done
+
+			it 'should return no tags for user 1', (done) =>
+				TagsClient.getUserTags @user_id_1, (err, res, body) =>
+					tags = body.map((tagDoc) -> tagDoc.name)
+					expect(tags).to.deep.equal([])
+					done()
+
+			it 'should return the two tags for user 2', (done) =>
+				TagsClient.getUserTags @user_id_2, (err, res, body) =>
+					tags = body.map((tagDoc) -> tagDoc.name).sort()
+					expect(tags).to.deep.equal([@tag_1, @tag_2])
+					done()
+
+		describe 'renameTag', =>
+			beforeEach (done) =>
+				@tag_1_renamed = "#{@tag_1}_renamed__"
+				TagsClient.renameTag @user_id_1, @user_1_tag_1_id, @tag_1_renamed, done
+
+			it 'should return the new name for tag 1', (done) =>
+				TagsClient.getUserTags @user_id_1, (err, res, body) =>
+					tags = body.map((tagDoc) -> tagDoc.name).sort()
+					expect(tags[0]).to.equal(@tag_1_renamed)
+					done()
+
+		describe 'adding a project to a tag', =>
```

Comment:
One good thing to test for here would be when adding a project to a tag, that adding a second project does not overwrite the first one.

One Mongo gotcha is that you could use `$set` instead of `$push` and accidentally clear out your old projects.